_posts/2020-08-05-#19160.md

```diff
@@ -7,17 +7,47 @@ authors: [ryanofsky]
+
+In followup PR [#10102 Multiprocess bitcoin](https://github.com/bitcoin/bitcoin/pull/10102), this functionality is used so a `bitcoin-gui` process can spawn a `bitcoin-node` process, and a `bitcoin-node` process can spawn a `bitcoin-wallet` process, and gui, node, and wallet functionality can run in separate processes that are protected from each other. In further followups [#19460 Add bitcoin-wallet -ipcconnect option](https://github.com/bitcoin/bitcoin/pull/19460) and [#19461 Add bitcoin-gui -ipcconnect option](https://github.com/bitcoin/bitcoin/pull/19461), more flexibility is added so node, wallet, and GUI processes can be started and stopped independently.
+
+All communication between process happens though [internal C++ interfaces](https://github.com/bitcoin/bitcoin/tree/master/src/interfaces), which are just C++ classes with pure virtual methods. The virtual methods allow bitcoin GUI, node, and wallet code to be written in a straightforward way that doesn't require dealing with complications of IPC (interprocess communication). IPC is handled by the multiprocess framework added in this PR, to avoid the need to complicate application code with low level I/O.
+
+Specifically, followup PR [#10102 Multiprocess bitcoin](https://github.com/bitcoin/bitcoin/pull/10102) uses the IPC framework added here to generate subclasses for each C++ interface class, with every overridden virtual method of every subclass implemented to send method calls and arguments to a remote process, wait for a response, and then return the response as the method return value. For example, if the GUI wants to find out if a wallet address is spendable, it calls the [`interfaces::Wallet::isSpendable()`](https://github.com/ryanofsky/bitcoin/blob/pr/ipc.120/src/interfaces/wallet.h#L94-L95) method. If GUI code and wallet code are running in the same process, this directly invokes the [`interfaces::WalletImpl::isSpendable()` method implementation](https://github.com/ryanofsky/bitcoin/blob/pr/ipc.120/src/interfaces/wallet.cpp#L135). But if GUI and wallet code are running in different process, the multiprocess framework instead provides a different (generated) [`interfaces::Wallet`](https://github.com/ryanofsky/bitcoin/blob/pr/ipc.120/src/interfaces/wallet.h#L51-L52) class implementation, that forwards the method request and arguments to a remote wallet process, waits for the results, and returns them, instead of directly calling wallet code.
```

Comment:
s/, that forwards/, which forwards| that forwards/ (either "which" with comma, or "that" without)